db/sg_replicate_cfg.go

```diff
@@ -832,11 +832,20 @@ func (m *sgReplicateManager) UpsertReplication(replication *ReplicationUpsertCon
 		_, exists := cluster.Replications[replication.ID]
 		if exists {
 			created = false
+			// If replication already exists ensure its in the stopped state before allowing upsert
+			state, err := m.GetReplicationStatus(replication.ID, DefaultReplicationStatusOptions())
+			if err != nil {
+				return true, err
+			}
+			if state.Status != ReplicationStateStopped {
```

Comment:
As discussed, do a manual test (at minimum) to ensure that if another goroutine starts the replication after we've checked the status but prior to updating the config, then the update to the config fails with a CAS error, and the retry fails (because state.Status is running).

Reply:
Manually tested as mentioned above.
Just put a temporary sleep after the check and during that sleep started the replication again. Still hits the error case.